File: Converted_POS_08May2025/Converted_POS/Models/clsCourseCategory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Data;
 
using System.Data.SqlClient;
using System.Linq;
using System.Threading.Tasks;

namespace Converted_POS.Models
{
    public class clsCourseCategory
    {
        public int? course_category_id { get; set; }
        public int cmp_id { get; set; }
        public bool is_active { get; set; }
        public DateTime? created_date { get; set; }
        public DateTime? modify_date { get; set; }
        public String Tran_Type { get; set; }
        [Required]
        public string name { get; set; }
        public String active { get; set; }

    }

    //public class clsCourseCategory_DAL
    //{
    //To View all  details    
    //public IEnumerable<clsCourseCategory> SelectAll(int? c_id, string conn)
    //{
    //    List<clsCourseCategory> list = new List<clsCourseCategory>();

    //    using (SqlConnection con = new SqlConnection(conn))
    //    {
    //        SqlCommand cmd = new SqlCommand("Get_M_Course_Category", con);
    //        cmd.CommandType = CommandType.StoredProcedure;

    //        cmd.Parameters.AddWithValue("@cmp_id", c_id);

    //        con.Open();
    //        SqlDataReader rdr = cmd.ExecuteReader();

    //        while (rdr.Read())
    //        {
    //            clsCourseCategory cat = new clsCourseCategory();

    //            cat.course_category_id = Convert.ToInt32(rdr["course_category_id"]);
    //            cat.name = rdr["Name"].ToString();
    //            cat.active = rdr["Active"].ToString();

    //            if (rdr["Active"].ToString() == "Yes")
    //            {
    //                cat.is_active = true;
    //            }
    //            else
    //            {
    //                cat.is_active = false;
    //            }

    //            list.Add(cat);
    //        }
    //        con.Close();
    //    }
    //    return list;
    //}

    //Get the details of a particular Category  
    //public clsCourseCategory Select(int? c_id, int? id, string conn)
    //{
    //    clsCourseCategory cat = new clsCourseCategory();

    //    using (SqlConnection con = new SqlConnection(conn))
    //    {
    //        SqlCommand cmd = new SqlCommand("Get_M_Course_CategoryInfo", con);
    //        cmd.CommandType = CommandType.StoredProcedure;

    //        cmd.Parameters.AddWithValue("@course_category_id", id);
    //        cmd.Parameters.AddWithValue("@cmp_id", c_id);

    //        con.Open();
    //        SqlDataReader rdr = cmd.ExecuteReader();

    //        while (rdr.Read())
    //        {
    //            cat.course_category_id = Convert.ToInt32(rdr["course_category_id"]);
    //            cat.name = rdr["Name"].ToString();
    //            cat.active = rdr["is_active"].ToString();

    //            if (rdr["is_active"].ToString() == "1")
    //            {
    //                cat.is_active = true;
    //            }
    //            else
    //            {
    //                cat.is_active = false;
    //            }
    //        }

    //        con.Close();
    //    }
    //    return cat;
    //}

    //To Add new Category record    
    //public void Insert(clsCourseCategory cat, string conn)
    //{
    //    using (SqlConnection con = new SqlConnection(conn))
    //    {
    //        SqlCommand cmd = new SqlCommand("P_M_Course_Category", con);
    //        cmd.CommandType = CommandType.StoredProcedure;

    //        cmd.Parameters.AddWithValue("@cmp_id", cat.cmp_id);
    //        cmd.Parameters.AddWithValue("@course_category_id ", cat.course_category_id);
    //        cmd.Parameters.AddWithValue("@name", cat.name);
    //        cmd.Parameters.AddWithValue("@is_active", cat.is_active);
    //        cmd.Parameters.AddWithValue("@Tran_Type", "I");

    //        con.Open();
    //        cmd.ExecuteNonQuery();
    //        con.Close();
    //    }
    //}

    //To Update Category record    
    //public void Update(clsCourseCategory cat, string conn)
    //{
    //    using (SqlConnection con = new SqlConnection(conn))
    //    {
    //        SqlCommand cmd = new SqlCommand("P_M_Course_Category", con);
    //        cmd.CommandType = CommandType.StoredProcedure;

    //        cmd.Parameters.AddWithValue("@cmp_id", cat.cmp_id);
    //        cmd.Parameters.AddWithValue("@course_category_id ", cat.course_category_id);
    //        cmd.Parameters.AddWithValue("@name", cat.name);
    //        cmd.Parameters.AddWithValue("@is_active", cat.is_active);
    //        cmd.Parameters.AddWithValue("@Tran_Type", "U");

    //        con.Open();
    //        cmd.ExecuteNonQuery();
    //        con.Close();
    //    }
    //}

    //To Delete Category record    
    //public void Delete(clsCourseCategory cat, string conn)
    //{
    //    using (SqlConnection con = new SqlConnection(conn))
    //    {
    //        SqlCommand cmd = new SqlCommand("P_M_Course_Category", con);
    //        cmd.CommandType = CommandType.StoredProcedure;

    //        cmd.Parameters.AddWithValue("@cmp_id", cat.cmp_id);
    //        cmd.Parameters.AddWithValue("@course_category_id ", cat.course_category_id);
    //        cmd.Parameters.AddWithValue("@name", cat.name);
    //        cmd.Parameters.AddWithValue("@is_active", cat.is_active);
    //        cmd.Parameters.AddWithValue("@Tran_Type", "D");

    //        con.Open();
    //        cmd.ExecuteNonQuery();
    //        con.Close();
    //    }
    //}

    //}

}

File: POS_10Apr2025/POS/App_Code/clsCourseCategory.vb

Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Globalization
Imports System.Collections.Generic
Imports System.Threading
Public Class Controller_clsCourseCategory


    Inherits BaseClass
    Dim oClsDal As ClsDataccess = New ClsDataccess()
    Public Sub New()
    End Sub
#Region "Public Variables"
    Private _course_category_id As Integer
    Private _cmp_id As Integer
    Private _is_active As Byte
    Private _Course_id As String
    Private _created_date As System.DateTime
    Private _modify_date As System.DateTime
    Private _Tran_Type As String
    Private _name As String
    Private objclscoursecategory As clsCourseCategory
#End Region
    Public Property course_category_id() As Integer
        Get
            Return _course_category_id
        End Get
        Set(ByVal value As Integer)
            _course_category_id = value
        End Set
    End Property
    Public Property cmp_id() As Integer
        Get
            Return _cmp_id
        End Get
        Set(ByVal value As Integer)
            _cmp_id = value
        End Set
    End Property
    Public Property Course_id() As String
        Get
            Return _Course_id
        End Get
        Set(ByVal value As String)
            _Course_id = value
        End Set
    End Property
    Public Property created_date() As System.DateTime
        Get
            Return _created_date

        End Get
        Set(ByVal value As System.DateTime)
            _created_date = value
        End Set
    End Property
    Public Property modify_date() As System.DateTime
        Get
            Return _modify_date

        End Get
        Set(ByVal value As System.DateTime)
            _modify_date = value
        End Set
    End Property
    Public Property is_active() As Byte
        Get
            Return _is_active
        End Get
        Set(ByVal value As Byte)
            _is_active = value
        End Set
    End Property
    Public Property Tran_Type() As String
        Get
            Return _Tran_Type
        End Get
        Set(ByVal value As String)
            _Tran_Type = value
        End Set
    End Property
    Public Property Name() As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            _name = value
        End Set
    End Property
    Public Function Selectcourse() As DataTable
        Dim dt As DataTable
        Try
            objclscoursecategory = New clsCourseCategory()
            objclscoursecategory.cmp_id = cmp_id
            dt = objclscoursecategory.Selectcourse()
            Return dt
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Insert() As Boolean
        Try
            objclscoursecategory = New clsCourseCategory()
            objclscoursecategory.Name = Name
            objclscoursecategory.cmp_id = cmp_id
            objclscoursecategory.course_category_id = course_category_id
            objclscoursecategory.is_active = is_active
            If objclscoursecategory.Insert() Then
                Return True
            End If
            Return False
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Update() As Boolean
        Try
            objclscoursecategory = New clsCourseCategory()
            objclscoursecategory.Name = Name
            objclscoursecategory.cmp_id = cmp_id
            objclscoursecategory.course_category_id = course_category_id
            objclscoursecategory.is_active = is_active

            If objclscoursecategory.Update() Then
                Return True
            End If
            Return False
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Delete() As Boolean
        Try
            objclscoursecategory = New clsCourseCategory()
            objclscoursecategory.Name = Name
            objclscoursecategory.cmp_id = cmp_id
            objclscoursecategory.course_category_id = course_category_id
            objclscoursecategory.is_active = is_active
            If objclscoursecategory.Delete() Then
                Return True
            End If
            Return False
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function SelectAll() As DataTable
        Dim dt As DataTable
        Try
            objclscoursecategory = New clsCourseCategory()
            objclscoursecategory.cmp_id = cmp_id
            dt = objclscoursecategory.[SelectAll]()
            Return dt
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function

    Function [Select]() As DataTable
        Dim dt As DataTable
        Try
            objclscoursecategory = New clsCourseCategory()
            objclscoursecategory.course_category_id = course_category_id
            objclscoursecategory.cmp_id = cmp_id
            dt = objclscoursecategory.[Select]()
            Return dt
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
End Class
Public Class clsCourseCategory

    Inherits BaseClass
    Dim oClsDal As ClsDataccess = New ClsDataccess()
#Region "Public Variables"
    Private _course_category_id As Integer
    Private _cmp_id As Integer
    Private _is_active As Byte
    Private _Course_id As String
    Private _created_date As System.DateTime
    Private _modify_date As System.DateTime
    Private _Tran_Type As String
    Private _name As String
    Private objclscoursecategory As clsCourseCategory
#End Region

    Public Property course_category_id() As Integer
        Get
            Return _course_category_id
        End Get
        Set(ByVal value As Integer)
            _course_category_id = value
        End Set
    End Property
    Public Property cmp_id() As Integer
        Get
            Return _cmp_id
        End Get
        Set(ByVal value As Integer)
            _cmp_id = value
        End Set
    End Property
    Public Property Course_id() As String
        Get
            Return _Course_id
        End Get
        Set(ByVal value As String)
            _Course_id = value
        End Set
    End Property
    Public Property created_date() As System.DateTime
        Get
            Return _created_date

        End Get
        Set(ByVal value As System.DateTime)
            _created_date = value
        End Set
    End Property
    Public Property modify_date() As System.DateTime
        Get
            Return _modify_date

        End Get
        Set(ByVal value As System.DateTime)
            _modify_date = value
        End Set
    End Property
    Public Property is_active() As Byte
        Get
            Return _is_active
        End Get
        Set(ByVal value As Byte)
            _is_active = value
        End Set
    End Property
    Public Property Tran_Type() As String
        Get
            Return _Tran_Type
        End Get
        Set(ByVal value As String)
            _Tran_Type = value
        End Set
    End Property
    Public Property Name() As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            _name = value
        End Set
    End Property
    Public Function Selectcourse() As DataTable
        Dim ds As New DataSet
        Dim oColSqlparram As ColSqlparram = New ColSqlparram()
        oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
        Dim dtlogin As DataTable = oClsDal.GetdataTableSp("Get_Course", oColSqlparram)

        Return dtlogin
    End Function
    Public Function Insert() As Boolean
        Try
            Dim oColSqlparram As ColSqlparram = New ColSqlparram()
            oColSqlparram.Add("@name", name)
            oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
            oColSqlparram.Add("@course_category_id", course_category_id, SqlDbType.Int)
            oColSqlparram.Add("@is_active", is_active, SqlDbType.TinyInt)
            oColSqlparram.Add("@Tran_Type", "I")

            Dim dtlogin As DataTable = oClsDal.GetdataTableSp("P_M_Course_Category", oColSqlparram)
            Return True
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Update() As Boolean
        Try
            Dim oColSqlparram As ColSqlparram = New ColSqlparram()
            oColSqlparram.Add("@course_category_id", course_category_id, SqlDbType.Int)
            oColSqlparram.Add("@name", Name)
            oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
            oColSqlparram.Add("@course_category_id", course_category_id, SqlDbType.Int)
            oColSqlparram.Add("@is_active", is_active, SqlDbType.TinyInt)
            oColSqlparram.Add("@Tran_Type", "U")

            Dim dtlogin As DataTable = oClsDal.GetdataTableSp("P_M_Course_Category", oColSqlparram)
            Return True
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Delete() As Boolean
        Try
            Dim oColSqlparram As ColSqlparram = New ColSqlparram()
            oColSqlparram.Add("@course_category_id", course_category_id, SqlDbType.Int)
            oColSqlparram.Add("@name", Name)
            oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)

            oColSqlparram.Add("@is_active", is_active, SqlDbType.TinyInt)
            oColSqlparram.Add("@Tran_Type", "D")

            Dim dtlogin As DataTable = oClsDal.GetdataTableSp("P_M_Course_Category", oColSqlparram)
            Return True
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function [SelectAll]() As DataTable
        Dim ds As New DataSet

        Dim oColSqlparram As ColSqlparram = New ColSqlparram()
        oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)
        Dim dtlogin As DataTable = oClsDal.GetdataTableSp("Get_M_Course_Category", oColSqlparram)

        Return dtlogin
    End Function
    Public Function [Select]() As DataTable
        Dim ds As New DataSet

        Dim oColSqlparram As ColSqlparram = New ColSqlparram()
        oColSqlparram.Add("@course_category_id", course_category_id, SqlDbType.Int)
        oColSqlparram.Add("@cmp_id", cmp_id, SqlDbType.Int)


        Dim dtlogin As DataTable = oClsDal.GetdataTableSp("Get_M_Course_CategoryInfo", oColSqlparram)

        Return dtlogin
    End Function

End Class
